prov/efa/configure.m4

```diff
@@ -87,6 +87,14 @@ AC_DEFUN([FI_EFA_CONFIGURE],[
 	      ])
 	CPPFLAGS=$save_CPPFLAGS
 
+	AS_IF([test x"$enable_efa" != x"no"],
+	      [AC_CHECK_DECL(EFADV_DEVICE_ATTR_CAPS_RDMA_WRITE,
+			    [AC_DEFINE([HAVE_CAPS_RDMA_WRITE], [1], [HAVE_CAPS_RDMA_WRITE is defined])],
```

Comment:
The common practice is to always define the Macro, but set the value to 0 if the flag is not available.

Then use `if` instead of `ifdef` in the code

Reply:
This is not the case for the 2 preceding defines: `HAVE_CAPS_RNR_RETRY` and `HAVE_RDMA_SIZE`

Should I favor (local) consistency or `if`?

Reply:
> Should I favor (local) consistency or if?

This practise came from a bug that a macro defined as 0 was treated as a yes.

Reply:
I see.  That's a good reason.

I changed my commit, and added a new one to switch the other to defines to be used as you say.